index.js

```diff
@@ -1,58 +1,57 @@
-'use strict';
 const async = require('async');
 const defaultMethod = require('lodash.defaults');
 const unset = require('lodash.unset');
 const Boom = require('boom');
 const defaults = {};
 
-exports.register = (server, options, next) => {
+const register = async(server, options) => {
   options = defaultMethod(options, defaults);
   const getReplyHandler = (autoMethod, autoOptions) => {
     const legacy = (autoOptions.reply);
     let replyCalled = false;
-    return (request, reply) => {
+    return async (request, h) => new Promise(async(resolve, reject) => {
```

Comment:
I'm not sure this is right - you have async in here twice and you are returning a promise